sycl/source/detail/device_impl.cpp

```diff
@@ -222,6 +220,51 @@ pi_native_handle device_impl::getNative() const {
   return Handle;
 }
 
+bool device_impl::has(aspect Aspect) const {
+  switch (Aspect) {
+  case aspect::host:
+    return is_host();
+    break;
+  case aspect::cpu:
+    return is_cpu();
+    break;
+  case aspect::gpu:
+    return is_gpu();
+    break;
+  case aspect::accelerator:
+    return is_accelerator();
+    break;
+  case aspect::fp16:
+    return has_extension("cl_khr_fp16");
+    break;
+  case aspect::fp64:
+    return has_extension("cl_khr_fp64");
+    break;
+  case aspect::int64_base_atomics:
+    return has_extension("cl_khr_int64_base_atomics");
+    break;
+  case aspect::int64_extended_atomics:
+    return has_extension("cl_khr_int64_extended_atomics");
+    break;
+  case aspect::image:
+    return get_info<info::device::image_support>();
+    break;
+  case aspect::online_compiler:
+    return get_info<info::device::is_compiler_available>();
+    break;
+  case aspect::online_linker:
+    return get_info<info::device::is_linker_available>();
+    break;
+  case aspect::queue_profiling:
+    return get_info<info::device::queue_profiling>();
+    break;
+  default:
+    throw runtime_error("This device aspect has not been implemented yet.",
+                        PI_INVALID_DEVICE);
+  }
+  return false;
```

Comment:
I guess this is also unreachable code since the function either returns a value according to specified `aspect` or throws an exception. Is it possible to remove this?

Reply:
No, this is not unreachable code.  There are many more aspects that are not implemented yet.  This catches those cases.

Reply:
As far as I can see, in case of not implemented aspect an exception is thrown which ends function execution. So this `return` statement is redundant, IMO.

Reply:
removed return statement.